### PR TITLE
Display user profile

### PR DESCRIPTION
**Pull request by aniansson**
## Pivotal Tracker story 

https://www.pivotaltracker.com/story/show/138269297

## Changes proposed in this pull request:

Displays the user profile

## What I have learned working on this feature:
The basic skills of rails, routes, controllers

### CODE REVIEW

**Comment by aniansson** [comment]
Review and merge _**"Add dish to my dishes"**_ code first!!! Test failing due to missing stuff from that branch.

**Comment by tochman** [comment]
I ❤️ HAML

File: app/views/checkout/index.html.erb [hunk 1]
```diff
@@ -20,8 +20,9 @@
                     <img class="media-object" src="http://icons.iconarchive.com/icons/custom-icon-design/flatastic-2/72/product-icon.png" style="width: 72px; height: 72px;">
                   </a>
                   <div class="media-body">
-                    <h4 class="media-heading"><a href="#"><%= order_item.item.name.humanize %></a></h4>
-                    <h5 class="media-heading"> by <a href="#">username</a></h5>
+                    <h4 class="media-heading"><%= link_to "#{order_item.item.name.humanize}", "#", data: {toggle: "modal", target: "#dish#{order_item.item.id}Modal"} %></h4>
+                    <h5 class="media-heading">
```
Comment (tochman) [inline app/views/checkout/index.html.erb:24]: This way of naming the internal link seems odd. I will let this pass but `#dish2Modal` feels weird. We don't use camel case in Id's or anchor links

File: features/display_user_profile.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  I need to be able to display a profile page
+
+  Background:
+    Given that the following users exist
+      | first_name  | last_name | email               | address        | username |
+      | Philip      | Zudemberg | email@random.com    | Address 14     | Philippo |
+    And that there is a session with lat and lng near "Chalmers"
+
+  Scenario: Display of user profile
+    When I am on the "landing" page
```
Comment (tochman) [inline features/display_user_profile.feature]: Given. It should be Given

File: features/display_user_profile.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    And I click "Add dish"
+    And I fill in "Dish name" with "Meatballs"
+    And I fill in "Dish description" with "This are like your Swedish mom's homemade meatballs"
+    And I fill in "Price" with "59.99"
+    And I fill in "Nr of portions available" with "5"
+    And I fill in "Earliest pickup time" with "2017-02-17 18:00"
+    And I fill in "Pickup address" with "Storgatan 1, Stockholm"
+    And I click "Create dish"
+    And I should see "Successfully added 5 portions of Meatballs at $59.99 each to your dishes"
```
Comment (tochman) [inline features/display_user_profile.feature]: Then. Are you guys reading your own scenarios?

File: features/display_user_profile.feature [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    And I click "My profile"
+    Then I should see "email@random.com"
+    And I should see "Philip Zudemberg"
+    And I should see "Address 14"
+    And I click the button for "Meatballs"
+    And I should see "Meatballs"
+    And I should see "$59.99"
+    And I should see "18:00"
```
Comment (tochman) [inline features/display_user_profile.feature]: Way too long scenario.

File: features/step_definitions/display_dishes_steps.rb [hunk 5]
```diff
@@ -1,5 +1,7 @@
 Given(/^the following dishes exists$/) do |table|
   table.hashes.each do |hash|
-    FactoryGirl.create(:dish, hash)
+    @dish = FactoryGirl.create(:dish, hash)
   end
+  @dish.user = User.find_by(username: "Philippo")
+  @dish.save
```
Comment (tochman) [inline features/step_definitions/display_dishes_steps.rb]: I would do this. More readible
```
user = User.find_by(username: "Philippo")
@dish.update_attribute(user: user)
```

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/views/checkout/index.html.erb (hunk 1 above)
Comment (Philippoes) [reply app/views/checkout/index.html.erb:24]: Gotcha

**Review by tochman — APPROVED** [review]

**Pull request merged**